nasConvert/Time.go

```diff
@@ -12,24 +12,33 @@ func toBinaryCodedDecimal(val int) int {
 	return ((val / 10) << 4) + (val % 10)
 }
 
+// Refer to TS 23.040 - 9.1.2.3  Semi-octet representation
 func toSemiOctet(val int) int {
-	// Refer to TS 23.040 - 9.1.2.3  Semi-octet representation
 	return ((val & 0x0F) << 4) | ((val & 0xF0) >> 4)
 }
 
-func parseTimeZone(timezone string) int {
+func parseTimeZoneToNas(timezone string) int {
 	time := 0 // expressed in quarters of an hour
 
 	// Parse hour
 	if timezone[1] == '1' {
 		time += (10 * 4)
 	}
-
 	for i := 0; i < 10; i++ {
 		if int(timezone[2]) == (i + 0x30) {
 			time += i * 4
 		}
 	}
+	if strings.Contains(timezone, "+1") || strings.Contains(timezone, "+2") {
```

Comment:
Will this mistake e.g. +10:45 as a string containing DST?

Reply:
Yes, it will.